Source/AsInterface.cs

```diff
@@ -107,6 +101,8 @@ public override Switches Switches
 			set => this.owner.Switches = value;
 		}
 
+		internal override Type TargetType => this.owner.TargetType;
```

Comment:
This is *almost* identical to `MockedType`, except for the overrides in `AsInterface`.